worlds/sm64ex/Rules.py

```diff
@@ -122,4 +122,9 @@ def set_rules(world, player: int, area_connections):
     add_rule(world.get_location("MIPS 1", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS1Cost[player].value))
     add_rule(world.get_location("MIPS 2", player), lambda state: state.can_reach("Basement", 'Region', player) and state.has("Power Star", player, world.MIPS2Cost[player].value))
 
-    world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    if world.CompletionType[player] == "Last_Bowser_Stage":
+        world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
+    elif world.CompletionType[player] == "All_Bowser_Stages":
```

Comment:
```suggestion
    if world.CompletionType[player] == "last_bowser_stage":
        world.completion_condition[player] = lambda state: state.can_reach("Bowser in the Sky", 'Region', player)
    elif world.CompletionType[player] == "all_bowser_stages":
```

Turns out I gave wrong advice, this should actually work.